src/components/sidebar/nodeView/ShortIdInput.tsx

```diff
@@ -0,0 +1,164 @@
+import classnames from 'classnames';
+import { reaction, IReactionDisposer } from 'mobx';
+import { inject, observer } from 'mobx-react';
+import React from 'react';
+import { Button } from '~/components/controls';
+import Dropdown, { IDropdownItem } from '~/components/controls/Dropdown';
+import InputContainer from '~/components/controls/InputContainer';
+import ButtonType from '~/enums/buttonType';
+import { INode } from '~/models';
+import StopService from '~/services/stopService';
+import { NodeStore } from '~/stores/nodeStore';
+import * as s from './shortIdInput.scss';
+
+interface IStopFormProps {
+    node: INode;
+    isEditingDisabled: boolean;
+    nodeInvalidPropertiesMap: object;
+    nodeStore?: NodeStore;
+    onNodePropertyChange: (property: keyof INode) => (value: any) => void;
+}
+
+interface IStopFormState {
+    availableShortIds: string[];
+    availableShortIdDropdownItems: IDropdownItem[];
+    isAvailableShortIdsDropdownVisible: boolean;
+}
+
+@inject('nodeStore')
+@observer
+class ShortIdInput extends React.Component<IStopFormProps, IStopFormState> {
+    private mounted: boolean;
+    private isShortIdLetterListener: IReactionDisposer;
+
+    constructor(props: IStopFormProps) {
+        super(props);
+        this.state = {
+            availableShortIds: [],
+            availableShortIdDropdownItems: [],
+            isAvailableShortIdsDropdownVisible: false
+        };
+    }
+
+    componentDidMount() {
+        this.mounted = true;
```

Comment:
Needed because we have async functions before set state:
```
if (this.mounted) {
            this.setState({
                availableShortIds,
                availableShortIdDropdownItems: this.createAvailableShortIdDropdownItems(
                    availableShortIds
                )
            });
        }
```